admin_manual/configuration_server/bruteforce_configuration.rst

```diff
@@ -2,55 +2,205 @@
+
+.. danger::
+
+   Any excluded IP address can perform authentication attempts without any throttling.
+   It's best to exclude as few IP addresses as you can, or even none at all. 
+
+Additional Details
+------------------
+
+Feature History
+~~~~~~~~~~~~~~~
+
+(Notable)
+
+* Server 10.0:
+   - Added: Initial implementation
+   - Changed: Protection of major authentication entrypoints
+* Server 12.0:
+   - Added: Annotation support for protecting API calls
+   - Added: Distinct action types for more complex handling combinations
+   - Changed: Protection of some public APIs
+   - Changed: Protection of public links
+   - Fixed: Disregard existing recent attempts immediately if BPF is disabled manually
+   - Added: Admin security settings section for managing some parameters (via the ``bruteforcesettings`` shipped app)
+   - Added: Enable the admin to exclude IP addresses from throttling (via the ``bruteforcesettings`` shipped app)
+   - Changed: Controllers now must explicly call throttle() to increase throttle counter
+* Server 13.0
+   - Changed: Reset bruteforce attempts upon successful login (only the entries associated with that user from that IP)
+   - Added: Make bruteforce status available via Capabilities
+* Server 14.0
+   - Fixed: Truncate long usernames (>64 characters) during failed login attempts before storing the attempt entry
+   - Fixed: Decrease maxDelay from 30s to 25s to avoid unintentional client authentication timeouts
+* Server 15.0
+   - Changed: Reset bruteforce attempts upon successful token refresh when using OAuth
+* Server 18.0
+   - Fixed: Handle scoped IPv6 addresses
+* Server 20.0
+   - Changed: When maximum delay is reached and maximum attempts permitted have been exceeded within the past 30 minutes, return "429 Too Many Requests" until no longer the true
+   - Added: ``occ`` command to reset bruteforce attempts for an IP (``occ security:bruteforce:reset``)
+* Server 21.0
+   - Added: Once a day cronjob added to cleanup stale attempt entries
+   - Changed: Delete all entries >48 hours old
+* Server 24.0
+   - Added: Protection of password reset page
+   - Added: Logging of throttling or blocking (info level)
+* Server 25.0
+   - Added: Enable apps to reset the delay for an IP
+   - Added: Enable apps to get the current delay for an IP
+* Server 26.0
+   - Changed: Protection for public shares
+* Server 28.0
+   - Fixed: Reset bruteforce attempts upon successful sudo attempt
+   - Fixed: CSP errors when injecting ``X-Nextcloud-Bruteforce-Throttled`` (the throttle delay header)
+   - Added: Memcache based backend
+   - Added: Show admins when they are throttled
+   - Added: Show current user throttling status (via the ``bruteforcesettings`` app)
+* Server 29.0
+   - Fixed: Don't throw a "500 Internal Server Error" when MaxDelayReached; instead return a "429 Too Many Requests"
+   - Fixed: Prevent setting empty IP masks when adding an exclusion (in the ``bruteforcesetting`` app) 
```

Comment:
Not a fan of this block. It gives way too many details for attack vectors on old instances:

```suggestion
* Server 10:
   - Added: Initial implementation - Protecting major authentication entrypoints
* Server 12:
   - Added: Annotation support for protecting API calls
   - Added: Distinct action types for more complex handling combinations
   - Added: Enable the admin to exclude IP addresses from throttling (via the ``bruteforcesettings`` shipped app)
   - Changed: Controllers now must explicly call throttle() to increase throttle counter
* Server 13
   - Changed: Reset bruteforce attempts upon successful login (only the entries associated with that user from that IP)
   - Added: Make bruteforce status available via Capabilities
* Server 14
   - Fixed: Decrease maxDelay from 30s to 25s to avoid unintentional client authentication timeouts
* Server 18
   - Fixed: Handle scoped IPv6 addresses
* Server 20
   - Changed: When maximum delay is reached and maximum attempts permitted have been exceeded within the time frame, return "429 Too Many Requests"
   - Added: ``occ`` command to reset bruteforce attempts for an IP (``occ security:bruteforce:reset``)
* Server 24
   - Added: Logging of throttling or blocking (info level)
* Server 25
   - Added: Enable apps to reset the delay for an IP
   - Added: Enable apps to get the current delay for an IP
* Server 28
   - Added: Memcache based backend
   - Added: Show admins when they are throttled
   - Added: Show current IP throttling status (via the ``bruteforcesettings`` app)
```

but even then it is a mix of addressing admins and developers.

Reply:
Streamlined the history block: admin-only stuff now (removed as much as possible of the other + dev-only bits)